DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Avalonia.Media.Imaging;
+using Blast.API.Core.Processes;
+using Blast.API.Processes;
+using Blast.API.Search;
+using Blast.Core.Interfaces;
+using Blast.Core.Objects;
+using Blast.Core.Results;
+using TextCopy;
+using static DuckDuckGo.Fluent.Plugin.DuckDuckGoSearchResult;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.AppFunctions;
+using static DuckDuckGo.Fluent.Plugin.QrFunctions;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    internal class DuckDuckGoSearchApp : ISearchApplication
+    {
+        private const string SearchAppName = "DuckDuckGo Instant Answers";
+        private readonly SearchApplicationInfo _applicationInfo;
+        private BitmapImageResult _logoImage;
+
+        public DuckDuckGoSearchApp()
+        {
+            _applicationInfo = new SearchApplicationInfo(SearchAppName,
+                DuckTagDescription, SearchOperations)
+            {
+                MinimumSearchLength = 1,
+                IsProcessSearchEnabled = false,
+                IsProcessSearchOffline = false,
+                SearchTagOnly = true,
+                ApplicationIconGlyph = SearchResultIcon,
+                SearchAllTime = ApplicationSearchTime.Fast,
+                DefaultSearchTags = SearchTags
+            };
+        }
+
+        public SearchApplicationInfo GetApplicationInfo()
+        {
+            return _applicationInfo;
+        }
+
+        public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchRequest,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            string searchedTag = searchRequest.SearchedTag;
+            string searchedText = searchRequest.SearchedText;
+
+            if (!VerifySearchedTerms(searchedText, searchedTag)) yield break;
+            searchedText = searchedText.Trim();
+
+            if (!string.IsNullOrWhiteSpace(searchedTag) && searchedTag.Equals(QrTag))
```

Comment:
Instead of double checking the search tag, you can make the `VerifySearchedTerms` method return an Enum that tells the search what to do - break, search normally, or search QR code

Reply:
In latest code, I've used an Enum to decide the action i.e., to search normally, generate QR or break if Null.